src/domain/products/details/variants/variant-editor.tsx

```diff
@@ -0,0 +1,362 @@
+  console.log(variant)
+  const [currencyOptions, setCurrencyOptions] = useState([])
+  const [prices, setPrices] = useState(variant.prices)
+  const [selectedCountry, setSelectedCountry] = useState(
+    variant.origin_country
+      ? countryOptions.find((cd) => cd.label === variant.origin_country)
+      : undefined
+  )
+
+  console.log("selected country", selectedCountry)
+  const { setValue, getValues, register, reset, watch, handleSubmit } = useForm(
+    variant
+  )
+
+  useEffect(() => {
+    reset({
+      ...variant,
+    })
+
+    variant.options.forEach((option, index) => {
+      register(`options.${index}.option_id`)
+      setValue(`options.${index}.option_id`, option.option_id)
+    })
+  }, [variant])
+
+  const getCurrencyOptions = () => {
+    return ((store && store.currencies) || [])
+      .map((v) => ({
+        value: v.code.toUpperCase(),
+        label: v.code.toUpperCase(),
+      }))
+      .filter(
+        (o) => !prices.find((p) => !p.edit && p.currency_code === o.value)
+      )
+  }
+
+  useEffect(() => {
+    setCurrencyOptions(getCurrencyOptions())
+  }, [store, isLoading, variant.prices])
+
+  const handleCurrencySelected = (index, currency) => {
+    const newPrices = [...prices]
+    newPrices[index] = {
+      ...newPrices[index],
+      currency_code: currency.toLowerCase(),
+    }
+
+    setPrices(newPrices)
+  }
+
+  const handlePriceChange = (index, value) => {
+    console.log(value)
+    const newPrices = [...prices]
+    newPrices[index] = {
+      ...newPrices[index],
+      amount: value,
+    }
+
+    setPrices(newPrices)
+  }
+
+  const removePrice = (index) => {
+    const newPrices = [...prices]
+    newPrices.splice(index, 1)
+    setPrices(newPrices)
+  }
+
+  const addPrice = () => {
+    const newPrices = [
+      ...prices,
+      {
+        edit: true,
+        region: "",
+        currency_code: currencyOptions[0].value,
+        amount: "",
+        sale_amount: "",
+      },
+    ]
+
+    setPrices(newPrices)
+  }
+
+  const handleSave = (data) => {
+    console.log("submitted", data)
+
+    data.prices = prices.map(({ currency_code, region_id, amount }) => ({
+      currency_code,
+      region_id,
+      amount: Math.round(amount),
+    }))
+
+    data.origin_country = selectedCountry?.label
+
+    data.prices = data.prices.map((p) => removeNullish(p))
+    const cleaned = convertEmptyStringToNull(data, numberFields)
+
+    console.log("saved", cleaned)
+
+    onSubmit(cleaned)
+  }
+
+  watch()
+
+  // as="form"
+  return (
+    <Modal handleClose={onCancel}>
+      <Modal.Body>
+        <Modal.Header handleClose={onCancel}>
+          <h2 className="inter-xlarge-semibold">Edit Variant</h2>
+        </Modal.Header>
+        <Modal.Content>
+          <div className="mb-8">
+            <label className="inter-base-semibold mb-4 flex items-center gap-xsmall">
+              {"Prices"}
+              <InfoTooltip content={"Variant prices"} />
```

Comment:
This tooltip is acting a bit weird. It seems to default to its open state. Whenever I open a variant editor, I see the following: 
![image](https://user-images.githubusercontent.com/59018053/151935888-653f8fa6-83c4-4465-b57e-16290ffc1378.png)